gpiozero/internal_devices.py

```diff
@@ -514,51 +544,98 @@ class TimeOfDay(PolledInternalDevice):
         The time after which the device will be considered inactive.
 
     :param bool utc:
-        If :data:`True` (the default), a naive UTC time will be used for the
-        comparison rather than a local time-zone reading.
+        If `None` (the default), UTC time will be used for the comparison. 
+        If `False` the local clock time will be use, ignoring the timezone. 
+        (If `True` a naive UTC time will be used - this is not recommended, 
+        see the note above)
 
     :type event_delay: float
     :param event_delay:
-        The number of seconds between file reads (defaults to 10 seconds).
+        The number of seconds between file reads (defaults to 5 seconds).
 
     :type pin_factory: Factory or None
     :param pin_factory:
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
     """
-    def __init__(self, start_time, end_time, *, utc=True, event_delay=5.0,
+
+    def __init__(self, start_time, end_time, *, utc=None, event_delay=5.0,
                  pin_factory=None):
-        self._start_time = None
-        self._end_time = None
-        self._utc = True
+    
+        utc_deprecation_warning = (
+        'Using utc=True is deprecated and scheduled for removal in a future version.'
+        ' Start and end times will default to tzinfo=datetime.UTC unless you specify'
+        ' different tzinfo when defining them.'
+        )
+        if utc:
+            warnings.warn(utc_deprecation_warning, DeprecationWarning)
+
+        self._aware = utc is None
```

Comment:
_Perhaps_ we don't need a separate `self._aware` attribute, and the `timezone_aware` property could just `return self._utc is None` ?

Reply:
I had thought about that, and chosen to leave this in __init__ as I feel it is a lot more explicit than having the logic later in the code as part of the property. As a bonus (but possible YAGNI) this is also where any logic around a `local` argument would be placed, and used to define self._aware.
I'll rename this to _tzaware to be a bit more explicit.